solidity/contracts/maintainer/MaintainerProxy.sol

```diff
@@ -0,0 +1,588 @@
+
+        bridge.notifyMovingFundsBelowDust(walletPubKeyHash, mainUtxo);
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + _notifyMovingFundsBelowDustGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Used by the wallet to prove the BTC moved funds sweep
+    ///         transaction and to make the necessary state changes. Moved
+    ///         funds sweep is only accepted if it satisfies SPV proof.
+    ///
+    ///         The function validates the sweep transaction structure by
+    ///         checking if it actually spends the moved funds UTXO and the
+    ///         sweeping wallet's main UTXO (optionally), and if it locks the
+    ///         value on the sweeping wallet's 20-byte public key hash using a
+    ///         reasonable transaction fee. If all preconditions are
+    ///         met, this function updates the sweeping wallet main UTXO, thus
+    ///         their BTC balance.
+    ///
+    ///         It is possible to prove the given sweep transaction only
+    ///         one time.
+    /// @param sweepTx Bitcoin sweep funds transaction data
+    /// @param sweepProof Bitcoin sweep funds proof data
+    /// @param mainUtxo Data of the sweeping wallet's main UTXO, as currently
+    ///        known on the Ethereum chain
+    function submitMovedFundsSweepProof(
+        BitcoinTx.Info calldata sweepTx,
+        BitcoinTx.Proof calldata sweepProof,
+        BitcoinTx.UTXO calldata mainUtxo
+    ) external onlyMaintainer {
+        uint256 gasStart = gasleft();
+
+        bridge.submitMovedFundsSweepProof(sweepTx, sweepProof, mainUtxo);
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + _submitMovedFundsSweepProofGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Wraps request new wallet call and reimburses a caller's
+    ///         transaction cost.
+    /// @param activeWalletMainUtxo Data of the active wallet's main UTXO, as
+    ///        currently known on the Ethereum chain.
+    function requestNewWallet(BitcoinTx.UTXO calldata activeWalletMainUtxo)
+        external
+        onlyMaintainer
+    {
+        uint256 gasStart = gasleft();
+
+        bridge.requestNewWallet(activeWalletMainUtxo);
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + _requestNewWalletGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Notifies that the wallet is either old enough or has too few
+    ///         satoshis left and qualifies to be closed.
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    /// @param walletMainUtxo Data of the wallet's main UTXO, as currently
+    ///        known on the Ethereum chain.
+    function notifyCloseableWallet(
+        bytes20 walletPubKeyHash,
+        BitcoinTx.UTXO calldata walletMainUtxo
+    ) external onlyMaintainer {
+        uint256 gasStart = gasleft();
+
+        bridge.notifyCloseableWallet(walletPubKeyHash, walletMainUtxo);
```

Comment:
This function was recently renamed to `notifyWalletClosable`. We should change everything containing "ClosableWallet" to "WalletClosable" in this PR (functions, variables, descriptions).